### PR TITLE
[SP-4478] Backport of BACKLOG-22673 - When editing some hops of Job E…

### PR DESCRIPTION
**Pull request by smmribeiro**
…xecutor &  one of Transformation Executor appropriate xml tags related to hops are cleared (7.1 Suite)

@pentaho-lmartins @cravobranco @ricardosilva88

### CODE REVIEW

**Comment by ricardosilva88** [comment]
From my analysis, the base case had 2 PRs:
https://github.com/pentaho/pentaho-kettle/pull/5219/files
https://github.com/pentaho/pentaho-kettle/pull/5519/files

Please verify, in this PR your are back porting the 2nd case only

**Comment by smmribeiro** [comment]
@ricardosilva88: I've added the missing code.

File: engine/src/org/pentaho/di/trans/step/StepMetaInterface.java [hunk 1]
```diff
@@ -166,7 +166,7 @@
   /**
    * Set default values
    */
-  public void setDefault();
+  void setDefault();
```
Comment (ricardosilva88) [inline engine/src/org/pentaho/di/trans/step/StepMetaInterface.java]: Changing public access modifier to default it's making all the methods access visibility more restrictive. If some other code part or lib is depending from these methods and if it is outside of the package it'll cause problems. I would recommend to keep the modifier as it is.

Comment (ricardosilva88) [reply engine/src/org/pentaho/di/trans/step/StepMetaInterface.java]: It applies to all methods that you changed the modifier in this class, ty

**Pull request merged**